pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -513,7 +514,18 @@ func applyHTTPRouteDestination(
 				Substitution: regexRewrite.Rewrite,
 			}
 		} else if uri := in.Rewrite.GetUri(); uri != "" {
-			action.PrefixRewrite = uri
+			if model.UseGatewaySemantics(vs) && uri == "/" {
+				// remove the prefix
+				action.RegexRewrite = &matcher.RegexMatchAndSubstitute{
```

Comment:
AFAIK this is the same case as https://github.com/cilium/cilium/pull/28294 tries to solve, but they do it differently. We should probably find the best way and make sure implementations are aligned.

Reply:
I don't think [cilium/cilium#28294](https://github.com/cilium/cilium/pull/28294/files#diff-7c81c69ba8f86a61afee1d1eb36a4eb786f3c9251df6bc86edf6d14a476e5cfcR283) is correct:
```go
route.Route.RegexRewrite = &envoy_type_matcher_v3.RegexMatchAndSubstitute{
	Pattern: &envoy_type_matcher_v3.RegexMatcher{
		Regex: "^" + httpRoute.PathMatch.Prefix,
	},
	Substitution: "",
}
```
It does not handle the case where the entire path is removed. In this case we must set the URI to `/`. Please see https://www.rfc-editor.org/rfc/rfc9112.html#section-3.2.1-2.
> If the target URI's path component is empty, the client __MUST__ send "/" as the path within the origin-form of request-target.

If I have a `RegexRewrite` like:
```yaml
regexRewrite:
  pattern:
    regex: ^/foo
  substitution: ""
```
I will get 502 when accessing `/foo`

I will probably send a patch to cilium to fix that